static/js/click_handlers.js

```diff
@@ -578,6 +579,40 @@ export function initialize() {
         do_render_buddy_list_tooltip(elem.parent(), title_data);
     });
 
+    function do_render_status_emoji_name_tooltip(elem, title_data) {
+        let placement = "bottom";
+        if (window.innerWidth < media_breakpoints_num.md) {
+            // On small devices display tooltips based on available space.
+            // This will default to "bottom" placement for this tooltip.
+            placement = "auto";
+        }
+        tippy(elem[0], {
+            // Quickly display and hide status emoji name tooltips
+            // so that they don't stick and overlap with
+            // each other.
+            delay: 0,
+            content: render_status_emoji_name_tooltip_content(title_data),
+            arrow: true,
+            placement,
+            allowHTML: true,
+            showOnCreate: true,
+            onHidden: (instance) => {
+                instance.destroy();
+            },
+            appendTo: () => document.body,
```

Comment:
Do we need to specify all of these parameters?  

@amanagr may be able to help simplify this implementation.

Reply:
I can think of two extractions with `do_render_buddy_list_tooltip`.

One `user_list_tooltip_props`: Store the common properties between both of them there. I can count 5.
Second: Extract a function to calculate `placement` instead of duplicating the code.

Reply:
These can both be prep commints of their own.

Reply:
Thanks for the first extraction. Can you do the second one I mentioned too? Also, you should clean up your commit history by removing merge type commits.